src/gt4py/next/common.py

```diff
@@ -901,6 +915,18 @@ def __gt_builtin_func__(cls, _: fbuiltins.BuiltInFunction) -> Never:  # type: ig
     def ndarray(self) -> Never:
         raise NotImplementedError()
 
+    def __add__(self, other: Field | core_defs.IntegralScalar) -> Field:
+        if isinstance(other, int):
+            return CartesianConnectivity(dimension=self.codomain, offset=self.offset + other)
+        else:
+            raise TypeError("'ConnectivityField' does not support this operation.")
+
+    def __sub__(self, other: Field | core_defs.IntegralScalar) -> Field:
+        if isinstance(other, int):
+            return CartesianConnectivity(dimension=self.codomain, offset=self.offset - other)
+        else:
+            raise TypeError("'ConnectivityField' does not support this operation.")
```

Comment:
```suggestion
    def __add__(self, other: Field | core_defs.IntegralScalar) -> Field:
        assert isinstance(other, int)
        return CartesianConnectivity(dimension=self.codomain, offset=self.offset + other)

    def __sub__(self, other: Field | core_defs.IntegralScalar) -> Field:
        assert isinstance(other, int)
        return CartesianConnectivity(dimension=self.codomain, offset=self.offset - other)
```

Should be moved to CartesianConnectivity class. An even cleaner solution that would avoid issues when the attributes of CartesianConnectivity change would be to use [dataclasses.replace](https://docs.python.org/3/library/dataclasses.html):

```
dataclasses.replace(self, offset=self.offset+other)
```